test/watcherEx.test.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018 The Casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { newEnforcer } from 'casbin';
+import { RedisWatcherEx } from '../src/watcherEx';
+
+test('Test single WatcherEx on Redis', async() => {
+  const myCallbackFuncE1 = async (sec: string, ptype: string, params: string[]) => {
+    await e1.addPolicyInternal(sec, ptype, params);
+  }
+  const watcherEx = await RedisWatcherEx.newWatcherEx('redis://localhost:6379/5', myCallbackFuncE1);
+  const e1 = await newEnforcer('examples/authz_model.conf', 'examples/authz_policy.csv');
+  const e2 = await newEnforcer('examples/authz_model.conf', 'examples/authz_policy.csv');
+  e1.setWatcherEx(watcherEx);
+  e2.setWatcherEx(watcherEx);
+  await e2.addPolicy('alice', '/dataset2/resource2', 'POST');
+  expect(await e2.enforce('alice', '/dataset2/resource2', 'POST')).toBe(true);
+  expect(await e1.enforce('alice', '/dataset2/resource2', 'POST')).toBe(true);
```

Comment:
@nodece I found out that the callback is running after this line, that's why the test is failing, what can I do to solve it ?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

Reply:
Construct the test differently. Easiest way to do it is to use `done`-syntax or if you prefer async, wrap it within a promise like [here](https://github.com/Sefriol/casbin-mongo-changestream-watcher/blob/master/test/integration/watcher.test.ts#L30). Then you can move expects within the callback-function and resolve it after all tests are done.